pkg/config/orderer.go

```diff
@@ -333,3 +333,38 @@ func AddOrdererEndpoint(config *cb.Config, orgName string, endpoint string) erro
 
 	return nil
 }
+
+// RemoveOrdererEndpoint removes an orderer's endpoint from an existing channel config transaction.
+// The removed endpoint and org it belongs to must both already exist.
+func RemoveOrdererEndpoint(config *cb.Config, orgName string, endpoint string) error {
+	ordererOrgGroup, ok := config.ChannelGroup.Groups[OrdererGroupKey].Groups[orgName]
+	if !ok {
+		return fmt.Errorf("orderer org %s does not exist in channel config", orgName)
+	}
+
+	ordererAddrProto := &cb.OrdererAddresses{}
+
+	if ordererAddrConfigValue, ok := ordererOrgGroup.Values[EndpointsKey]; ok {
+		err := proto.Unmarshal(ordererAddrConfigValue.Value, ordererAddrProto)
+		if err != nil {
+			return fmt.Errorf("failed unmarshalling orderer org %s's endpoints: %v", orgName, err)
+		}
+	}
+
+	existingEndpoints := ordererAddrProto.Addresses
+	for i, e := range existingEndpoints {
```

Comment:
This is a bit of an unconventional loop for removing. Nothing wrong, per se, but it does mean we exit the function at an error and that we have a normal return in the middle.

Reply:
Example: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating